frontend/src/v5/store/tickets/ticketsGroups.helpers.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { isString } from 'lodash';
+import { ViewpointGroup, ViewpointGroupHierarchy, ViewpointState, Properties } from './tickets.types';
+
+const hierarchyWithGroups = (groups: Record<string, ViewpointGroup>) => (hierarchy: ViewpointGroupHierarchy) => {
+	const hierarchyToReturn = { ...hierarchy };
```

Comment:
`ViewpointGroupHierarchy` contains "prefix" and "colors" which are array, so they are passed as references.
The line above only does a shallow clone, so `hierarchyToReturn` will point to the same `color` address as the original `hierarchy`.

This may lead to subtle bugs in the future.